scripts/mixins/families/puk.lua

```diff
@@ -0,0 +1,45 @@
+-- Puk family mixin
+require("scripts/globals/mixins")
+require("scripts/globals/status")
+require("scripts/globals/world")
+-----------------------------------
+
+g_mixins = g_mixins or {}
+g_mixins.families = g_mixins.families or {}
+
+g_mixins.families.puk = function(mob)
+    mob:addListener("SPAWN", "PUK_SPAWN", function(puk)
+        puk:setMod(xi.mod.WIND_ABSORB, 100)
+    end)
+
+    mob:addListener("TAKE_DAMAGE", "PUK_TAKE_DAMAGE", function(puk, amount, attacker, attackType, damageType)
+        local elements =
+        {
+            {xi.damageType.FIRE,      xi.day.FIRESDAY},
+            {xi.damageType.EARTH,     xi.day.EARTHSDAY},
+            {xi.damageType.WATER,     xi.day.WATERSDAY},
+            {xi.damageType.WIND,      xi.day.WINDSDAY},
+            {xi.damageType.ICE,       xi.day.ICEDAY},
+            {xi.damageType.LIGHTNING, xi.day.LIGHTNINGSDAY},
+            {xi.damageType.LIGHT,     xi.day.LIGHTSDAY},
+            {xi.damageType.DARK,      xi.day.DARKSDAY}
+        }
+
+        -- If the element corresponding to the elemental day of the in-game Vana'diel week is used on a Puk, it will get 100% TP instantly.
+        for k, v in pairs(elements) do
+            if damageType == v[1] and VanadielDayOfTheWeek() == v[2] then
+                puk:addTP(3000)
+            end
+        end
+    end)
+
+    mob:addListener("ROAM_TICK", "PUK_ROAM_TICK", function(puk)
+        if VanadielDayOfTheWeek() == xi.day.WINDSDAY and puk:getMod(xi.mod.REGAIN) == 0 then
```

Comment:
They do this in wind weather. Anecdotal but feels like it doubles up in double weather.

Can't say they don't also get it from day, but I've not noticed it and wiki's don't mention it. They do mention wind weather though.

Reply:
Found another paragraph of info farther down old wiki's family page:
> Auto-Regain: On Lightsday and Windsday, and during Light or Wind Weather, Puks have the ability to regenerate TP, much like the Eruca of Mount Zhayolm.

so both day and weather, and alegedly light as well as wind but that has a verification tag so we can omit that pending an in game test on retail.

Reply:
On lightsday, Puks do not get Regain

Reply:
looks like we got day of week wind twice, one just needs to be weather and the pr is good

Reply:
oh I missed that one was for removing

Need an `or` in them then